python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -60,15 +60,18 @@ def get_nodes_to_launch(self, nodes: List[NodeID],
         """
 
         if resource_demands is None:
-            logger.info("No resource demands")
-            return []
+            node_resources, node_type_counts = self.calculate_node_resources(
+                nodes, pending_nodes, usage_by_ip)
+            nodes = _add_min_workers_nodes(self.node_types, node_type_counts,
+                                           nodes)
```

Comment:
Instead of calling _add_min in two places, can we move this function's current contents to `_get_nodes_to_launch_based_on_demands()`, and then call _add_min after that?